### PR TITLE
change(prometheus): redesign the latency metrics & update grafana

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

### What this PR does / why we need it:

Update point:
1. Prometheus metric: add upstream latency, rename latency -> request latency, overhead -> apisix latency
2. Grafana: move legend to the right with sort on total
3. Grafana: split status code
4. Grafana: delete batch process metric since it's optional

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by Yiyiyimu** [comment]
Waiting to update new screenshots

**Comment by Yiyiyimu** [comment]
To make it clear, put the updated images here
1-nginx
2-bandwidth
3-http
4-Misc

File: docs/en/latest/plugins/prometheus.md [hunk 1]
```diff
@@ -151,27 +153,17 @@ Or you can goto [Grafana official](https://grafana.com/grafana/dashboards/11719)
...
 * `Batch process entries`: A gauge type, when we use plugins and the plugin used batch process to send data, such as: sys logger, http logger, sls logger, tcp logger, udp logger and zipkin, then the entries which hasn't been sent in batch process will be counted in the metrics.
-* `Latency`: The per service histogram of request time and the overhead added by APISIX (request time - upstream response time).
+* `Latency`: The per service histogram of request time in different dimensions.
 
     Attributes：
 
     | Name      | Description |
     | ----------| ------------- |
-    | type      | Its value is fixed as `request`, which means HTTP request. |
+    | type      | The optional values are `apisix`, `upstream` and `request`, which means http latency caused by apisix, upstream, and in total. |
```
Comment (spacewander) [inline docs/en/latest/plugins/prometheus.md:162]: "The value can be `apisix`, `upstream` or `request`, which means http latency caused by apisix, upstream, or their sum." would be better?

Comment (Yiyiyimu) [reply docs/en/latest/plugins/prometheus.md]: Thanks for your suggestions and it is indeed more fluent! Fixed

**Review by spacewander — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**